demosplan/DemosPlanCoreBundle/Logic/FileResponseGenerator/ZipResponseGenerator.php

```diff
@@ -0,0 +1,119 @@
+    public function __invoke(array $file): Response
+    {
+        return $this->createStreamedResponseForZip($file);
+    }
+
+    private function createStreamedResponseForZip(array $file): StreamedResponse
+    {
+        return $this->zipExportService->buildZipStreamResponse(
+            $file['zipFileName'].'.zip',
+            fn (ZipStream $zipStream) => $this->fillZipWithData($zipStream, $file)
+        );
+    }
+
+    private function fillZipWithData(ZipStream $zipStream, array $file): void
+    {
+        $errorMessages = [];
+        $temporaryFullFilePath = DemosPlanPath::getTemporaryPath($file['xlsx']['filename']);
+        /** @var IWriter $writer */
+        $writer = $file['xlsx']['writer'];
+        try {
+            $writer->save($temporaryFullFilePath);
+            $this->zipExportService->addFileToZipStream(
+                $temporaryFullFilePath,
+                $file['zipFileName'].'/'.$file['xlsx']['filename'],
+                $zipStream
+            );
+        } catch (FileNotFoundException|FileNotReadableException $e) {
+            $this->logger->critical('Unable to load or read file from path.', [$e]);
+            $errorMessages[] = $this->translator->trans('error.statements.zip.export.generic.xlsx');
+        } catch (WriterException|Exception $e) {
+            $this->logger->critical('An error occured during creation of zip file for export.', [$e]);
+            $errorMessages[] = $this->translator->trans('error.statements.zip.export.generic.xlsx');
+        }
+
+        foreach ($file['attachments'] as $attachmentArray) {
+            /** @var File $attachment */
+            foreach ($attachmentArray as $attachment) {
+                try {
+                    $this->zipExportService->addFileToZipStream(
+                        $attachment->getFilePathWithHash(),
+                        $file['zipFileName'].'/'.$attachment->getHash().'_'.$attachment->getFilename(),
+                        $zipStream
+                    );
+                } catch (FileNotFoundException|FileNotReadableException $e) {
+                    $this->logger->critical('Unable to load or read file from path.', [$e]);
+                    $errorMessages[] = $this->translator->trans(
+                        'error.statments.zip.export.filenotfoundorreadable',
+                        ['hash' => $attachment->getHash()]
+                    );
+                } catch (InvalidDataException $e) {
+                    $this->logger->critical(
+                        'The hash for a file, which should be added to a zip export is not a string,
+            is an empty string, equals \'..\' or contains a slash (\'/\').',
+                        [$e]
+                    );
+                    $errorMessages[] = $this->translator->trans('error.statments.zip.export.hash.invalid');
+                } catch (Exception $e) {
+                    $this->logger->critical('An error occured during creation of zip file for export.', [$e]);
```

Comment:
typo occurred

Reply:
adjusted in [a158e92](https://github.com/demos-europe/demosplan-core/pull/2337/commits/a158e92d04f1a88c13b79ec1408522d8a3fba2fe)